pkg/cmd/util/clientcmd/gating.go

```diff
@@ -1,64 +1,49 @@
 package clientcmd
 
 import (
-	"encoding/json"
 	"fmt"
 
-	"github.com/blang/semver"
-
 	"github.com/openshift/origin/pkg/client"
-	"github.com/openshift/origin/pkg/version"
 )
 
-// Gate returns an error if the server is below minServerVersion or above/equal maxServerVersion.
-// To test only for min or only max version, set the other string to the empty value.
-func Gate(ocClient *client.Client, minServerVersion, maxServerVersion string) error {
-	if len(minServerVersion) == 0 && len(maxServerVersion) == 0 {
-		return fmt.Errorf("No version info passed to gate command")
+// Return err if the server does not support any of the legacy policy objects (< 3.7)
+func LegacyPolicyResourceGate(ocClient *client.Client) error {
```

Comment:
Needs to start with `LegacyPolicyResourceGate`.  Turn on the inspection in Gogland.

Reply:
Can this be changed to take the `client.Interface` instead?  Then you can actually add some unit tests.

Reply:
You may be able to take just the discovery interface, which would be preferred.